pycbc/inference/models/gaussian_noise.py

```diff
@@ -852,7 +856,59 @@ def _nowaveform_loglr(self):
             setattr(self._current_stats, '{}_optimal_snrsq'.format(det), 0.)
         return -numpy.inf
 
-    def _loglr(self):
+    def update(self, **params):
+        # update
+        super().update(**params)
+        # reset current waveforms
+        self._current_wfs = None
+
+    @property
+    def multi_signal_support(self):
+        """ The list of classes that this model supports in a multi-signal
+        likelihood
+        """
+        return [type(self)]
+
+    def multi_loglikelihood(self, models):
+        """ Calculate a multi-model (signal) likelihood
+        """
+        # Generate the waveforms for each submodel
+        wfs = []
+        for m in models:
+            wfs.append(m.get_waveforms())
+
+        # combine into a single waveform
+        combine = {}
+        for det in self.data:
+            mlen = max([len(x[det]) for x in wfs])
+            [x[det].resize(mlen) for x in wfs]
+            combine[det] = sum([x[det] for x in wfs])
+
+        loglr = self._loglr(waveforms=combine)
```

Comment:
I'd rather not change the API from the base models for the core likelihood functions. You don't need to do that anyway. Call self's get_waveforms, then add to the `current_wfs` the other models' waveforms. Inside `_loglr`, `get_waveforms` will be called again. Since `_current_wfs` will have been set, it will return the summed waveforms. You'll need to reset `_current_wfs` to None after, so that subsquent calls to the normal `loglr` (if someone were to do that) regenerates just this model's waveforms.